{"ast":null,"code":"import _classCallCheck from \"C:\\\\Users\\\\Rebekah\\\\Desktop\\\\Phase 2 Nanodegree\\\\final-project\\\\node_modules\\\\@babel\\\\runtime/helpers/esm/classCallCheck\";\nimport _createClass from \"C:\\\\Users\\\\Rebekah\\\\Desktop\\\\Phase 2 Nanodegree\\\\final-project\\\\node_modules\\\\@babel\\\\runtime/helpers/esm/createClass\";\nimport _possibleConstructorReturn from \"C:\\\\Users\\\\Rebekah\\\\Desktop\\\\Phase 2 Nanodegree\\\\final-project\\\\node_modules\\\\@babel\\\\runtime/helpers/esm/possibleConstructorReturn\";\nimport _getPrototypeOf from \"C:\\\\Users\\\\Rebekah\\\\Desktop\\\\Phase 2 Nanodegree\\\\final-project\\\\node_modules\\\\@babel\\\\runtime/helpers/esm/getPrototypeOf\";\nimport _inherits from \"C:\\\\Users\\\\Rebekah\\\\Desktop\\\\Phase 2 Nanodegree\\\\final-project\\\\node_modules\\\\@babel\\\\runtime/helpers/esm/inherits\";\nvar _jsxFileName = \"C:\\\\Users\\\\Rebekah\\\\Desktop\\\\Phase 2 Nanodegree\\\\final-project\\\\src\\\\components\\\\ErrorCatch.js\";\nimport React, { Component } from 'react';\n\nvar ErrorCatch =\n/*#__PURE__*/\nfunction (_Component) {\n  _inherits(ErrorCatch, _Component);\n\n  function ErrorCatch() {\n    _classCallCheck(this, ErrorCatch);\n\n    return _possibleConstructorReturn(this, _getPrototypeOf(ErrorCatch).apply(this, arguments));\n  }\n\n  _createClass(ErrorCatch, [{\n    key: \"componentDidCatch\",\n    value: function componentDidCatch() {\n      // Display fallback UI\n      this.setState({\n        error: true\n      });\n    }\n  }, {\n    key: \"render\",\n    value: function render() {\n      if (this.props.error) {\n        return React.createElement(\"h1\", {\n          __source: {\n            fileName: _jsxFileName,\n            lineNumber: 12\n          },\n          __self: this\n        }, \"Oh No! Something went wrong... Please try again later.\");\n      }\n\n      return this.props.children;\n    }\n  }]);\n\n  return ErrorCatch;\n}(Component);\n\nexport default ErrorCatch;","map":{"version":3,"sources":["C:\\Users\\Rebekah\\Desktop\\Phase 2 Nanodegree\\final-project\\src\\components\\ErrorCatch.js"],"names":["React","Component","ErrorCatch","setState","error","props","children"],"mappings":";;;;;;AAAA,OAAOA,KAAP,IAAgBC,SAAhB,QAAiC,OAAjC;;IAEMC,U;;;;;;;;;;;;;wCAEgB;AAClB;AACA,WAAKC,QAAL,CAAc;AAAEC,QAAAA,KAAK,EAAE;AAAT,OAAd;AACD;;;6BAEQ;AACP,UAAI,KAAKC,KAAL,CAAWD,KAAf,EAAsB;AACpB,eAAO;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA,oEAAP;AACD;;AACD,aAAO,KAAKC,KAAL,CAAWC,QAAlB;AACD;;;;EAZsBL,S;;AAezB,eAAeC,UAAf","sourcesContent":["import React, { Component } from 'react'\r\n\r\nclass ErrorCatch extends Component {\r\n\r\n  componentDidCatch() {\r\n    // Display fallback UI\r\n    this.setState({ error: true });\r\n  }\r\n\r\n  render() {\r\n    if (this.props.error) {\r\n      return <h1>Oh No! Something went wrong... Please try again later.</h1>;\r\n    }\r\n    return this.props.children;\r\n  }\r\n}\r\n\r\nexport default ErrorCatch"]},"metadata":{},"sourceType":"module"}